File: src/code/cpp/atoi.cpp
class Solution {
public:
    int myAtoi(string s) {
        int i=0, ans=0, n=s.length(), sign = 1;
        while(s[i]==' ') i++;
        if (s[i] == '-' or s[i] == '+')
        {
            sign = s[i] == '+' ? 1 : -1;
            i++;
        }
        while(isdigit(s[i]) && i<n){
            if(sign > 0){
                if(ans>INT_MAX/10 || (ans==INT_MAX/10 && s[i]-'0'>=7)){
                    return INT_MAX;
                }
            }else{
                if(ans>INT_MAX/10 || (ans==INT_MAX/10 && s[i]-'0'>=8)){
                    return INT_MIN;
                }
            }
            ans=ans*10+(s[i]-'0');
            i++;
        }
        ans *= sign;
        return ans;
    }
};

File: src/code/python/atoi.py
class Solution:
    def myAtoi(self, s: str) -> int:
        sign = 1
        start = 0
        ans = 0
        limit = 1 << 31
        s = s.strip(" ").split(" ")[0]
        size = len(s)
        if size == 0 : return 0
        if s[0] == '-':
            start = 1
            sign = -1
        if s[0] == '+':
            start = 1
        for i in range(start, size):
            if(s[i].isalpha() or s[i] =='.' or s[i] == '-' or s[i] == '+'): break
            ans = ans*10 + int(s[i])
        ans *= sign
        if ans > limit - 1:
            ans = limit - 1
        if ans < -(limit - 1):
            ans = -(limit - 1) -1
        return ans